src/main/haskell/language-kore/src/Data/Kore/ASTUtils/Substitution.hs

```diff
@@ -0,0 +1,124 @@
+{-|
```

Comment:
Is it necessary to implement substitution ourselves, or could we outsource this to a library like [bound](https://hackage.haskell.org/package/bound)?

Reply:
Hmm. Maybe. There was already a substitution function in Kore/Substitution, but I needed one that substitutes terms for terms and not just terms for variables.